core/extractors/extractors_code.py

```
@@ -0,0 +1,192 @@
+            '.sh': '#',
+            '.yaml': '#',
+            '.yml': '#'
+        }
+        
+        # Initialize Tree-sitter extractor
+        self.use_tree_sitter = use_tree_sitter
+        if use_tree_sitter:
+            try:
+                self.tree_sitter = TreeSitterExtractor()
+                logger.info("Initialized CodeExtractor with Tree-sitter support")
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)